drivers/azure/azureutil/azureutil.go

```diff
@@ -978,3 +995,21 @@ func extractStorageAccountFromVHDURL(vhdURL string) string {
 	}
 	return parts[0]
 }
+
+func BuildInstanceTags(tagGroups string) map[string]*string {
+	tags := make(map[string]*string)
+	if tagGroups == "" {
+		return tags
+	}
+
+	allTags := strings.Split(tagGroups, ",")
+	if len(allTags)%2 != 0 {
+		fmt.Printf("Tags are not in key value pairs. %d elements found\n", len(allTags))
+	}
+
+	for i := 0; i < len(allTags)-1; i += 2 {
+		tags[allTags[i]] = &allTags[i+1]
+	}
+
+	return tags
+}
```

Comment:
Should this be an error? Would it be better to err on the side of caution if the tags are incorrectly added? For example if a value accidentally contains a comma, and we end up shifting the remaining keys and values for tags? If so I'd make the following change:

```suggestion
func BuildInstanceTags(tagGroups string) (map[string]*string, error) {
	if tagGroups == "" {
		return nil, nil
	}

	allTags := strings.Split(tagGroups, ",")
	if len(allTags)%2 != 0 {
		return nil, fmt.Errorf("Tags are not in key value pairs. %d elements found\n", len(allTags))
	}
	
	tags := make(map[string]*string, len(allTags)/2))
	for i := 0; i < len(allTags); i += 2 { // don't need -1 since allTags must be properly formatted
		tags[allTags[i]] = &allTags[i+1]
	}

	return tags
}
```

If not, then don't worry about it, I'm going to approve.

Reply:
This is also how the EC2 driver handles tags, so for consistency I think we should just keep it for now. I have some pending work I want to do for the EC2 tag implementation for regular and spot instances, so I think I will change it in both drivers when I make those changes